lib/network/kmsAWS/Client.ts

```diff
@@ -0,0 +1,256 @@
+
+            if (!data) {
+                const error = _arsenalError("failed to generate data key: empty response");
+                logger.error("AWS KMS: failed to generate data key: empty response");
+                cb(error);
+                return;
+            }
+
+            logger.debug("AWS KMS: data key generated");
+            cb(null, Buffer.from(data.Plaintext as Uint8Array), Buffer.from(data.CiphertextBlob as Uint8Array));
+        });
+    }
+
+    cipherDataKey(
+        cryptoScheme: number,
+        masterKeyId: string,
+        plainTextDataKey: Buffer,
+        logger: werelogs.Logger,
+        cb: (err: Error | null, cipheredDataKey?: Buffer) => void
+    ): void {
+        logger.debug("AWS KMS: ciphering data key", { cryptoScheme, masterKeyId });
+        assert.strictEqual(cryptoScheme, 1);
+
+        const params = {
+            KeyId: masterKeyId,
+            Plaintext: plainTextDataKey,
+        };
+
+        this.client.encrypt(params, (err: AWSError, data) => {
+            if (err) {
+                const error = _arsenalError(err);
+                logger.error("AWS KMS: failed to cipher data key", { err });
+                cb(error);
+                return;
+            }
+
+            if (!data) {
+                const error = _arsenalError("failed to cipher data key: empty response");
+                logger.error("AWS KMS: failed to cipher data key: empty response");
+                cb(error);
+                return;
+            }
+
+            logger.debug("AWS KMS: data key ciphered");
+            cb(null, Buffer.from(data.CiphertextBlob as Uint8Array));
+            return;
+        });
+    }
+
+    decipherDataKey(
+        cryptoScheme: number,
+        masterKeyId: string,
+        cipheredDataKey: Buffer,
+        logger: werelogs.Logger,
+        cb: (err: Error | null, plainTextDataKey?: Buffer) => void
+    ): void {
+        logger.debug("AWS KMS: deciphering data key", { cryptoScheme, masterKeyId });
+        assert.strictEqual(cryptoScheme, 1);
+
+        const params = {
+            CiphertextBlob: cipheredDataKey,
+        };
+
+        this.client.decrypt(params, (err: AWSError, data) => {
+            if (err) {
+                const error = _arsenalError(err);
+                logger.error("AWS KMS: failed to decipher data key", { err });
+                cb(error);
+                return;
+            }
+
+            if (!data) {
+                const error = _arsenalError("failed to decipher data key: empty response");
+                logger.error("AWS KMS: failed to decipher data key: empty response");
+                cb(error);
+                return;
+            }
+
+            logger.debug("AWS KMS: data key deciphered");
+            cb(null, Buffer.from(data.Plaintext as Uint8Array));
+        });
+    }
+
+    healthcheck(logger: werelogs.Logger, cb: (err: Error | null) => void): void {
+        this.createMasterKey(logger, (err, masterKeyId) => {
```

Comment:
The KMS healthcheck was previously disabled in the [healthcheckHandler.js](https://github.com/scality/cloudserver/blob/development/7.70/lib/utilities/healthcheckHandler.js#L43). It is likely that the healthcheck will need to be enabled moving forward. I agree with the proposed changes and have updated the configuration accordingly.

Reply:
I'm still not sure about using an actual endpoint. It means the AK SK used will need privileges to do a `ListKeys`.
And requests to KMS have quotas and pricing: 

- https://eu-north-1.console.aws.amazon.com/servicequotas/home/services/kms/quotas/L-8A08DAEA
- https://aws.amazon.com/kms/pricing/

The default quota is 500 req/s. If anything else on the account reach this quota on this endpoint, it might fail the healthcheck.

Pricing is `$0.03 per 10,000 requests` (after 20,000 free tier requests).

With our regular deep healthcheck, 1 cloudserver might incur a small cost.
Now if there are a lot of cloudservers deployed, this will incur a noticable usage for quota and pricing

Reply:
Yes, I'm not sure what the best approach is here yet; we need to think about it.
Meanwhile, I will comment it out to avoid any issues related to costs or permissions.